lib/auth-middleware/auth_middleware/__init__.py

```diff
@@ -85,71 +95,93 @@ def authenticate(group_name=None, internal_token_only=False):
+
+            if is_member_of_key in id_info:
+                is_member_of = id_info[is_member_of_key]
+                print("groups: " + str(is_member_of))
+                if is_member_of_key == is_member_of_claim and group_name not in is_member_of:
+                    logger.warning(
+                        "user is not a member of the group " + group_name)
+                    raise OAuthProblem('Invalid token')
+            else:
+                logger.warning(is_member_of_key +
+                               " field is not present in the ID Token")
                 raise OAuthProblem('Invalid token')
 
 # Checks that the request has the right secret for this. This call is used initially and assumes that
 # the header contains the x-api-key. This (trivially) returns true of the verification worked and
 # otherwise will return various other exit codes.
+
+
 def verify_secret(request):
     key = request.headers.get(rokwire_api_key_header)
     if not key:
-        logger.warning("Request missing the " + rokwire_api_key_header + " header")
-        raise OAuthProblem('Missing API Key')  # missing header means bad request
+        logger.warning("Request missing the " +
+                       rokwire_api_key_header + " header")
+        # missing header means bad request
+        raise OAuthProblem('Missing API Key')
     # Assumption is that the key is a comma separated list of uuid's
     # This simply turns it in to a list and iterates. If the supplied key is in this list, true is returned
     # Otherwise, an error is raised.
-    keys = os.getenv('ROKWIRE_API_KEY').strip().split(',')
+    # keys = os.getenv('ROKWIRE_API_KEY').strip().split(',')
+    keys = os.getenv('ROKWIRE_API_KEY').strip()
+
     for test_key in keys:
         if key == test_key.strip():  # just in case there are embedded blanks
             return True
-    raise OAuthProblem('Invalid API Key') # failed matching means unauthorized in this context.
+    # failed matching means unauthorized in this context.
+    raise OAuthProblem('Invalid API Key')
 
 
 def verify_apikey(key, required_scopes=None):
     if not key:
-        logger.warning("API key is missing the " + rokwire_api_key_header + " header")
+        logger.warning("API key is missing the " +
+                       rokwire_api_key_header + " header")
         raise OAuthProblem('Missing API Key')
     # Assumption is that the key is a comma separated list of uuid's
     # This simply turns it in to a list and iterates. If the supplied key is in this list, true is returned
     # Otherwise, an error is raised.
-    keys = os.getenv('ROKWIRE_API_KEY').strip().split(',')
-    for test_key in keys:
-        if key == test_key.strip():  # just in case there are embedded blanks
-            return {'token_valid': True}
-    else:
+    # keys = os.getenv('ROKWIRE_API_KEY').strip().split(',')
+    api_key = os.getenv('ROKWIRE_API_KEY',
+                        '')
+    # for test_key in keys:
```

Comment:
Same with the code here too. For backward compatibility, the commented out code needs to be reinstated as currently we are using list of API Keys in ROKWIRE_API_KEY.